### ISSUE TITLE
Replace wurstscript.exe

### ISSUE THREAD

**Issue by Cokemonkey11** [peq/WurstScript#539]
As discussed in https://github.com/peq/WurstScript/issues/536 and https://github.com/peq/WurstScript/issues/535 - wurstscript.exe is a pretty dumb wrapper (and we dont have source code for it)

See https://github.com/Cokemonkey11/wurst-jar-wrapper - can someone please code review?

Key questions:

* Is a toml file sufficient for JVM args?
* How best to integrate it? Git submodule?

**Comment by peq** [comment]
Yay, Rust!

Code looks good, just a few comments:

1. I would not hardcode the JVM options in the config file format. Maybe people want to try other configurations as well.

    Maybe you can also make the path to Java configurable in the toml-file in case someone does not want to use the environment variables to configure it. Also we might want to provide a download with a bundled JVM in the future, so it would be nice if we could specify a relative path to the java executable.

2. `let args = env::args().skip(1).collect::<Vec<String>>().join(" ");` not sure if that is a good idea. I think it is better to pass the different args separately when starting the subprocess. Maybe Rust has a method for that?

3. Would it be possible to pass on the exit value from Wurst?

For integration: It depends on the size of the executable. Maybe we can just copy the executable here, since it probably won't change often. Another option would be that you create a release of the binary on GitHub and we download it from there as part of our build script. I would not want to add Rust (or any other tool that has to be installed besides gradle and Java) as a dependency to the build process.

**Comment by Cokemonkey11** [comment]
Thanks! Can someone review? https://github.com/Cokemonkey11/wurst-jar-wrapper/commit/e447e2ef0797f60cb91287cc0837a46edd674f5f

> I would not hardcode the JVM options in the config file format. Maybe people want to try other configurations as well.

I'm not quite clear about the question, but the config file is read during runtime, so it's not hardcoded. About java arguments: if you just comment them out [from the toml file], then nothing is passed to the jar. In that case you can just pass your own `-Xmx512m` to the wrapper and it will forward to jar like normal args.

Rest of 1, 2, 3: I agree, done :)

Integration: I won't complain if you want to just drop in the exe.

```
$ du -h target/*/*.exe
3.7M    target/debug/wurst-jar-wrapper.exe
1.4M    target/release/wurst-jar-wrapper.exe
```

**Comment by peq** [comment]
I mean that you hardcode what JVM-options can be configured. I would just let the user specify what arguments to pass to the jvm (i.e. args before `-jar` in the command). There are a lot of other options besides memory limits, for example here is a list of some: http://www.oracle.com/technetwork/articles/java/vmoptions-jsp-140102.html

Regarding the codes there is one more things I would check:
Is it possible to call the exe when you are in a different directory? Is the path to `wurstscript.jar` relative to the current directory or relative to the location of the `exe`?

**Comment by Cokemonkey11** [comment]
On the JVM options: right, that makes sense. I've added a "java args" option in addition to the existing memory options. Is that alright, or do we prefer to drop the memory arguments entirely?

On the directory paths: it should check the local directory where you execute, so if you run `./some/other/path/wurstscript.exe`, it will look for jar and config file in `.`. I'm testing in MSYS2 so not 100% sure on that...

Can someone review? https://github.com/Cokemonkey11/wurst-jar-wrapper/commit/5df22ace46a071d3e9ae1985d22a9bb19be71aff

**Comment by peq** [comment]
Code looks good, but I have not tried to run it yet.

**Comment by Frotty** [comment]
works splendid. merged via c09dcc9
